### ISSUE TITLE
It is not possible to run tests on Saucelabs mobile simulator in JWP (JSON Wire Protocol) mode over the tunnel

### ISSUE THREAD

**Issue by Sam55555** [webdriverio/webdriverio#6957]
**Environment (please complete the following information):**
 - **WebdriverIO version:** [6.4.0]
 - **Mode:** [WDIO Testrunner]
 - **If WDIO Testrunner, running sync/async:** [sync]
 - **Node.js version:** [v12.16.1]
 - **NPM version:** [6.13.4]
 - **Browser name and version:** [Chrome]
 - **Platform name and version:** [Android 11]

**Config of WebdriverIO**

**Describe the bug**
Currently, Saucelabs only supports the WebDriver JSON wire protocol for testing browsers on mobile simulators. To run tests in JWP mode on Saucelabs, the capability should have the JWP standard. When using the JWP capability, @wdio/sauce-service adds 'sauce:options' with tunnel identifier, which causes Saucelabs to start the test in W3C mode. When the mobile test is started in W3C mode, the webdriverio commands that use releaseActions have errors on Saucelabs.
In the "isLegacy" check here isW3C is always true on mobile

https://github.com/webdriverio/webdriverio/blob/4458755caf24420e203745622ac9c781a80cf742/packages/wdio-sauce-service/src/utils.ts#L74
because isAppium check here:
https://github.com/webdriverio/webdriverio/blob/4458755caf24420e203745622ac9c781a80cf742/packages/wdio-utils/src/envDetector.ts#L14
is true

**To Reproduce**

unzip example
npm i
npx wdio wdio.conf.js
[saucelabs-w3c-jwp.zip](https://github.com/webdriverio/webdriverio/files/6576398/saucelabs-w3c-jwp.zip)

**Expected behavior**

**Log**
[0-0] 2021-06-01T12:29:00.881Z INFO webdriver: COMMAND releaseActions()
[0-0] 2021-06-01T12:29:00.881Z INFO webdriver: [DELETE] https://ondemand.saucelabs.com/wd/hub/session/[some session number]/actions
[0-0] 2021-06-01T12:29:01.104Z WARN webdriver: Request failed with status 404 due to The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource

> Please use the correct markdown

**Comment by wswebcreation** [comment]
@Sam55555, @christian-bromann  

This doesn't seem to be an issue with Sauce Labs but or with WDIO, or with Appium. The test fails on the `actions.release` command which is translated to calling 

`[HTTP] --> DELETE /session/b6ce0427-3829-48c8-930e-57e6551f478d/actions`

This is failing in Appium, even on a local machine, see my WDIO and Appium logs below

**Local WDIO logs**
```log
❯ npx wdio wdio.conf.js

Execution of 1 spec files started at 2021-06-02T09:16:21.304Z

2021-06-02T09:16:21.306Z DEBUG @wdio/utils:initialiseServices: initialise service "appium" as NPM package
2021-06-02T09:16:21.319Z INFO @wdio/cli:launcher: Run onPrepare hook
2021-06-02T09:16:21.320Z DEBUG @wdio/appium-service: Will spawn Appium process: appium --base-path / --relaxed-security --log ./appium.log --port 4724
2021-06-02T09:16:22.214Z DEBUG @wdio/appium-service: Appium started with ID: 24684
2021-06-02T09:16:22.215Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2021-06-02T09:16:22.216Z INFO @wdio/local-runner: Start worker 0-0 with arg: wdio.conf.js
[0-0] 2021-06-02T09:16:22.504Z INFO @wdio/local-runner: Run worker command: run
[0-0] 2021-06-02T09:16:22.508Z DEBUG @wdio/local-runner:utils: init remote session
[0-0] 2021-06-02T09:16:22.510Z INFO webdriverio: Initiate new session using the ./protocol-stub protocol
[0-0] RUNNING in Chrome - /test/specs/example.e2e.js
[0-0] 2021-06-02T09:16:22.725Z DEBUG @wdio/utils:initialiseServices: initialise service "appium" as NPM package
[0-0] 2021-06-02T09:16:22.731Z DEBUG @wdio/local-runner:utils: init remote session
[0-0] 2021-06-02T09:16:22.732Z INFO webdriverio: Initiate new session using the webdriver protocol
[0-0] 2021-06-02T09:16:22.733Z INFO webdriver: [POST] http://localhost:4724/session
[0-0] 2021-06-02T09:16:22.733Z INFO webdriver: DATA {
  capabilities: {
    alwaysMatch: {
      browserName: 'Chrome',
      'appium:deviceName': 'Pixel_3_10.0',
      'appium:orientation': 'PORTRAIT',
      'appium:platformVersion': '10.0',
      platformName: 'Android'
    },
    firstMatch: [ {} ]
  },
  desiredCapabilities: {
    browserName: 'Chrome',
    'appium:deviceName': 'Pixel_3_10.0',
    'appium:orientation': 'PORTRAIT',
    'appium:platformVersion': '10.0',
    platformName: 'Android'
  }
}
[0-0] 2021-06-02T09:16:28.822Z INFO webdriver: COMMAND navigateTo("https://the-internet.herokuapp.com/login")
[0-0] 2021-06-02T09:16:28.822Z INFO webdriver: [POST] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/url
[0-0] 2021-06-02T09:16:28.822Z INFO webdriver: DATA { url: 'https://the-internet.herokuapp.com/login' }
[0-0] 2021-06-02T09:16:32.180Z INFO webdriver: COMMAND findElement("css selector", "#username")
[0-0] 2021-06-02T09:16:32.180Z INFO webdriver: [POST] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element
[0-0] 2021-06-02T09:16:32.180Z INFO webdriver: DATA { using: 'css selector', value: '#username' }
[0-0] 2021-06-02T09:16:32.202Z INFO webdriver: RESULT {
  'element-6066-11e4-a52e-4f735466cecf': '0.305102702666354-1',
  ELEMENT: '0.305102702666354-1'
}
[0-0] 2021-06-02T09:16:32.210Z INFO webdriver: COMMAND elementClick("0.305102702666354-1")
[0-0] 2021-06-02T09:16:32.210Z INFO webdriver: [POST] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element/0.305102702666354-1/click
[0-0] 2021-06-02T09:16:32.284Z INFO webdriver: COMMAND performActions(<object>)
[0-0] 2021-06-02T09:16:32.284Z INFO webdriver: [POST] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
[0-0] 2021-06-02T09:16:32.284Z INFO webdriver: DATA { actions: [ { type: 'key', id: 'keyboard', actions: [Array] } ] }
[0-0] 2021-06-02T09:16:32.656Z INFO webdriver: COMMAND releaseActions()
2021-06-02T09:16:32.657Z INFO webdriver: [DELETE] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
[0-0] 2021-06-02T09:16:32.672Z DEBUG webdriver: request failed due to status 9
2021-06-02T09:16:32.673Z INFO webdriver: Retrying 1/3
2021-06-02T09:16:32.673Z INFO webdriver: [DELETE] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
[0-0] 2021-06-02T09:16:32.672Z WARN webdriver: Request failed with status 404 due to The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource
[0-0] 2021-06-02T09:16:32.688Z DEBUG webdriver: request failed due to status 9
2021-06-02T09:16:32.688Z INFO webdriver: Retrying 2/3
2021-06-02T09:16:32.688Z INFO webdriver: [DELETE] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
[0-0] 2021-06-02T09:16:32.688Z WARN webdriver: Request failed with status 404 due to The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource
[0-0] 2021-06-02T09:16:32.692Z DEBUG webdriver: request failed due to status 9
[0-0] 2021-06-02T09:16:32.692Z INFO webdriver: Retrying 3/3
2021-06-02T09:16:32.692Z INFO webdriver: [DELETE] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
[0-0] 2021-06-02T09:16:32.692Z WARN webdriver: Request failed with status 404 due to The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource
[0-0] 2021-06-02T09:16:32.694Z DEBUG webdriver: request failed due to status 9
[0-0] 2021-06-02T09:16:32.694Z ERROR webdriver: Request failed with status 404 due to unknown command: The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource
[0-0] Error in "My Login application should login with valid credentials"
unknown command: The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource
[0-0] 2021-06-02T09:16:32.700Z INFO webdriver: COMMAND deleteSession()
[0-0] 2021-06-02T09:16:32.700Z INFO webdriver: [DELETE] http://localhost:4724/session/48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02T09:16:33.252Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code 1
[0-0] FAILED in Chrome - /test/specs/example.e2e.js
2021-06-02T09:16:33.253Z INFO @wdio/cli:launcher: Run onComplete hook
2021-06-02T09:16:33.253Z DEBUG @wdio/appium-service: Appium (pid: 24681) killed

 "spec" Reporter:
------------------------------------------------------------------
[emulator-5554 Android 10 #0-0] Spec: /Users/wimselles/Sauce/Git/customers/saucelabs-w3c-jwp/test/specs/example.e2e.js
[emulator-5554 Android 10 #0-0] Running: emulator-5554 on Android 10 executing Chrome
[emulator-5554 Android 10 #0-0] Session ID: 48c98e90-2d93-469b-8e30-cb555a1fc56a
[emulator-5554 Android 10 #0-0]
[emulator-5554 Android 10 #0-0] My Login application
[emulator-5554 Android 10 #0-0]    ✖ should login with valid credentials
[emulator-5554 Android 10 #0-0]
[emulator-5554 Android 10 #0-0] 1 failing (4.3s)
[emulator-5554 Android 10 #0-0]
[emulator-5554 Android 10 #0-0] 1) My Login application should login with valid credentials
[emulator-5554 Android 10 #0-0] unknown command: The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource
[emulator-5554 Android 10 #0-0]     at <Jasmine>
[emulator-5554 Android 10 #0-0]     at processTicksAndRejections (internal/process/task_queues.js:93:5)
[emulator-5554 Android 10 #0-0]     at LoginPage.login (/Users/wimselles/Sauce/Git/customers/saucelabs-w3c-jwp/test/pageobjects/login.page.js:20:17)
[emulator-5554 Android 10 #0-0]     at UserContext.<anonymous> (/Users/wimselles/Sauce/Git/customers/saucelabs-w3c-jwp/test/specs/example.e2e.js:8:19)
[emulator-5554 Android 10 #0-0]
[emulator-5554 Android 10 #0-0] Check out job at https://app.eu-central-1.saucelabs.com/tests/48c98e90-2d93-469b-8e30-cb555a1fc56a


Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:11 

2021-06-02T09:16:33.255Z INFO @wdio/local-runner: Shutting down spawned worker
2021-06-02T09:16:33.510Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2021-06-02T09:16:33.510Z INFO @wdio/local-runner: shutting down
```

**Appium Logs**
```log
2021-06-02 09:16:22:199 [Appium] Welcome to Appium v1.21.0
2021-06-02 09:16:22:200 [Appium] Non-default server args:
2021-06-02 09:16:22:200 [Appium]   port: 4724
2021-06-02 09:16:22:201 [Appium]   basePath: /
2021-06-02 09:16:22:201 [Appium]   logFile: ./appium.log
2021-06-02 09:16:22:201 [Appium]   relaxedSecurityEnabled: true
2021-06-02 09:16:22:214 [Appium] Appium REST http interface listener started on 0.0.0.0:4724
2021-06-02 09:16:22:765 [HTTP] --> POST /session
2021-06-02 09:16:22:765 [HTTP] {"capabilities":{"alwaysMatch":{"browserName":"Chrome","appium:deviceName":"Pixel_3_10.0","appium:orientation":"PORTRAIT","appium:platformVersion":"10.0","platformName":"Android"},"firstMatch":[{}]},"desiredCapabilities":{"browserName":"Chrome","appium:deviceName":"Pixel_3_10.0","appium:orientation":"PORTRAIT","appium:platformVersion":"10.0","platformName":"Android"}}
2021-06-02 09:16:22:766 [W3C] Calling AppiumDriver.createSession() with args: [{"browserName":"Chrome","appium:deviceName":"Pixel_3_10.0","appium:orientation":"PORTRAIT","appium:platformVersion":"10.0","platformName":"Android"},null,{"alwaysMatch":{"browserName":"Chrome","appium:deviceName":"Pixel_3_10.0","appium:orientation":"PORTRAIT","appium:platformVersion":"10.0","platformName":"Android"},"firstMatch":[{}]}]
2021-06-02 09:16:22:767 [BaseDriver] Event 'newSessionRequested' logged at 1622625382766 (11:16:22 GMT+0200 (Central European Summer Time))
2021-06-02 09:16:22:771 [Appium] 
2021-06-02 09:16:22:771 [Appium] ======================================================================
2021-06-02 09:16:22:771 [Appium]   DEPRECATION WARNING:
2021-06-02 09:16:22:771 [Appium] 
2021-06-02 09:16:22:771 [Appium]   The 'automationName' capability was not provided in the desired 
2021-06-02 09:16:22:772 [Appium]   capabilities for this Android session
2021-06-02 09:16:22:772 [Appium] 
2021-06-02 09:16:22:772 [Appium]   Setting 'automationName=UiAutomator2' by default and using the 
2021-06-02 09:16:22:772 [Appium]   UiAutomator2 Driver
2021-06-02 09:16:22:772 [Appium] 
2021-06-02 09:16:22:772 [Appium]   The next major version of Appium (2.x) will **require** the 
2021-06-02 09:16:22:772 [Appium]   'automationName' capability to be set for all sessions on all 
2021-06-02 09:16:22:772 [Appium]   platforms
2021-06-02 09:16:22:772 [Appium] 
2021-06-02 09:16:22:772 [Appium]   In previous versions (Appium <= 1.13.x), the default was 
2021-06-02 09:16:22:772 [Appium]   'automationName=UiAutomator1'
2021-06-02 09:16:22:772 [Appium] 
2021-06-02 09:16:22:772 [Appium]   If you wish to use that automation instead of UiAutomator2, please 
2021-06-02 09:16:22:773 [Appium]   add 'automationName=UiAutomator1' to your desired capabilities
2021-06-02 09:16:22:773 [Appium] 
2021-06-02 09:16:22:773 [Appium]   For more information about drivers, please visit 
2021-06-02 09:16:22:773 [Appium]   http://appium.io/docs/en/about-appium/intro/ and explore the 
2021-06-02 09:16:22:773 [Appium]   'Drivers' menu
2021-06-02 09:16:22:773 [Appium] 
2021-06-02 09:16:22:773 [Appium] ======================================================================
2021-06-02 09:16:22:773 [Appium] 
2021-06-02 09:16:22:958 [Appium] Appium v1.21.0 creating new AndroidUiautomator2Driver (v1.65.0) session
2021-06-02 09:16:22:959 [Appium] Applying relaxed security to 'AndroidUiautomator2Driver' as per server command line argument. All insecure features will be enabled unless explicitly disabled by --deny-insecure
2021-06-02 09:16:22:960 [BaseDriver] W3C capabilities and MJSONWP desired capabilities were provided
2021-06-02 09:16:22:960 [BaseDriver] Creating session with W3C capabilities: {
2021-06-02 09:16:22:960 [BaseDriver]   "alwaysMatch": {
2021-06-02 09:16:22:960 [BaseDriver]     "browserName": "Chrome",
2021-06-02 09:16:22:960 [BaseDriver]     "platformName": "Android",
2021-06-02 09:16:22:961 [BaseDriver]     "appium:deviceName": "Pixel_3_10.0",
2021-06-02 09:16:22:961 [BaseDriver]     "appium:orientation": "PORTRAIT",
2021-06-02 09:16:22:961 [BaseDriver]     "appium:platformVersion": "10.0"
2021-06-02 09:16:22:961 [BaseDriver]   },
2021-06-02 09:16:22:961 [BaseDriver]   "firstMatch": [
2021-06-02 09:16:22:961 [BaseDriver]     {}
2021-06-02 09:16:22:961 [BaseDriver]   ]
2021-06-02 09:16:22:961 [BaseDriver] }
2021-06-02 09:16:22:966 [BaseDriver] Session created with session id: 48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:22:967 [UiAutomator2] We're going to run a Chrome-based session
2021-06-02 09:16:22:967 [UiAutomator2] Chrome-type package and activity are com.android.chrome and com.google.android.apps.chrome.Main
2021-06-02 09:16:22:967 [UiAutomator2] Starting 'com.android.chrome' directly on the device
2021-06-02 09:16:22:982 [ADB] Found 13 'build-tools' folders under '/Users/wimselles/Library/Android/sdk' (newest first):
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/31.0.0-rc4
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/30.0.3
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/30.0.2
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/30.0.1
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/30.0.0-rc4
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/30.0.0-rc2
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/29.0.3
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/29.0.2
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/29.0.1
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/29.0.0
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/29.0.0-rc3
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/28.0.3
2021-06-02 09:16:22:983 [ADB]     /Users/wimselles/Library/Android/sdk/build-tools/27.0.3
2021-06-02 09:16:22:984 [ADB] Using 'adb' from '/Users/wimselles/Library/Android/sdk/platform-tools/adb'
2021-06-02 09:16:22:984 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 start-server'
2021-06-02 09:16:22:993 [AndroidDriver] Retrieving device list
2021-06-02 09:16:22:994 [ADB] Trying to find a connected android device
2021-06-02 09:16:22:994 [ADB] Getting connected devices
2021-06-02 09:16:23:001 [ADB] Connected devices: [{"udid":"emulator-5554","state":"device"}]
2021-06-02 09:16:23:002 [AndroidDriver] Looking for a device with Android '10.0.0'
2021-06-02 09:16:23:002 [ADB] Setting device id to emulator-5554
2021-06-02 09:16:23:002 [ADB] Getting device platform version
2021-06-02 09:16:23:002 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.release'
2021-06-02 09:16:23:016 [ADB] Current device property 'ro.build.version.release': 10
2021-06-02 09:16:23:017 [AndroidDriver] Using device: emulator-5554
2021-06-02 09:16:23:018 [ADB] Using 'adb' from '/Users/wimselles/Library/Android/sdk/platform-tools/adb'
2021-06-02 09:16:23:018 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 start-server'
2021-06-02 09:16:23:024 [ADB] Setting device id to emulator-5554
2021-06-02 09:16:23:025 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.sdk'
2021-06-02 09:16:23:038 [ADB] Current device property 'ro.build.version.sdk': 29
2021-06-02 09:16:23:038 [ADB] Getting device platform version
2021-06-02 09:16:23:038 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell getprop ro.build.version.release'
2021-06-02 09:16:23:050 [ADB] Current device property 'ro.build.version.release': 10
2021-06-02 09:16:23:050 [ADB] Device API level: 29
2021-06-02 09:16:23:050 [UiAutomator2] Relaxing hidden api policy
2021-06-02 09:16:23:051 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell 'settings put global hidden_api_policy_pre_p_apps 1;settings put global hidden_api_policy_p_apps 1;settings put global hidden_api_policy 1''
2021-06-02 09:16:23:094 [AndroidDriver] No app sent in, not parsing package/activity
2021-06-02 09:16:23:095 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 wait-for-device'
2021-06-02 09:16:23:101 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell echo ping'
2021-06-02 09:16:23:112 [AndroidDriver] Pushing settings apk to device...
2021-06-02 09:16:23:112 [ADB] Getting install status for io.appium.settings
2021-06-02 09:16:23:113 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package io.appium.settings'
2021-06-02 09:16:23:129 [ADB] 'io.appium.settings' is installed
2021-06-02 09:16:23:130 [ADB] Getting package info for 'io.appium.settings'
2021-06-02 09:16:23:130 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package io.appium.settings'
2021-06-02 09:16:23:163 [ADB] The version name of the installed 'io.appium.settings' is greater or equal to the application version name ('3.3.0' >= '3.3.0')
2021-06-02 09:16:23:164 [ADB] There is no need to install/upgrade '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/io.appium.settings/apks/settings_apk-debug.apk'
2021-06-02 09:16:23:164 [ADB] Getting IDs of all 'io.appium.settings' processes
2021-06-02 09:16:23:164 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell 'pgrep --help; echo $?''
2021-06-02 09:16:23:180 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
2021-06-02 09:16:23:199 [AndroidDriver] io.appium.settings is already running. There is no need to reset its permissions.
2021-06-02 09:16:23:200 [Logcat] Starting logs capture with command: /Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 logcat -v threadtime
2021-06-02 09:16:23:219 [UiAutomator2] Forwarding UiAutomator2 Server port 6790 to local port 8200
2021-06-02 09:16:23:220 [ADB] Forwarding system: 8200 to device: 6790
2021-06-02 09:16:23:220 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 forward tcp:8200 tcp:6790'
2021-06-02 09:16:23:237 [ADB] Getting install status for io.appium.uiautomator2.server
2021-06-02 09:16:23:238 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package io.appium.uiautomator2.server'
2021-06-02 09:16:23:262 [ADB] 'io.appium.uiautomator2.server' is installed
2021-06-02 09:16:23:262 [ADB] Getting package info for 'io.appium.uiautomator2.server'
2021-06-02 09:16:23:263 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package io.appium.uiautomator2.server'
2021-06-02 09:16:23:295 [ADB] The version name of the installed 'io.appium.uiautomator2.server' is greater or equal to the application version name ('4.21.1' >= '4.21.1')
2021-06-02 09:16:23:295 [UiAutomator2] io.appium.uiautomator2.server installation state: sameVersionInstalled
2021-06-02 09:16:23:296 [ADB] Checking app cert for /Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-v4.21.1.apk
2021-06-02 09:16:23:297 [ADB] Using 'apksigner.jar' from '/Users/wimselles/Library/Android/sdk/build-tools/31.0.0-rc4/lib/apksigner.jar'
2021-06-02 09:16:23:298 [ADB] Starting apksigner: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java -Xmx1024M -Xss1m -jar /Users/wimselles/Library/Android/sdk/build-tools/31.0.0-rc4/lib/apksigner.jar verify --print-certs /Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-v4.21.1.apk
2021-06-02 09:16:23:761 [ADB] apksigner stdout: Signer #1 certificate DN: EMAILADDRESS=android@android.com, CN=Android, OU=Android, O=Android, L=Mountain View, ST=California, C=US
2021-06-02 09:16:23:761 [ADB] Signer #1 certificate SHA-256 digest: a40da80a59d170caa950cf15c18c454d47a39b26989d8b640ecd745ba71bf5dc
2021-06-02 09:16:23:761 [ADB] Signer #1 certificate SHA-1 digest: 61ed377e85d386a8dfee6b864bd85b0bfaa5af81
2021-06-02 09:16:23:761 [ADB] Signer #1 certificate MD5 digest: e89b158e4bcf988ebd09eb83f5378e87
2021-06-02 09:16:23:761 [ADB] 
2021-06-02 09:16:23:761 [ADB] sha256 hash did match for 'appium-uiautomator2-server-v4.21.1.apk'
2021-06-02 09:16:23:762 [ADB] '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-v4.21.1.apk' is signed with the default certificate
2021-06-02 09:16:23:762 [ADB] Getting install status for io.appium.uiautomator2.server.test
2021-06-02 09:16:23:762 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package io.appium.uiautomator2.server.test'
2021-06-02 09:16:23:781 [ADB] 'io.appium.uiautomator2.server.test' is installed
2021-06-02 09:16:23:781 [ADB] Checking app cert for /Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-debug-androidTest.apk
2021-06-02 09:16:23:781 [ADB] Starting apksigner: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java -Xmx1024M -Xss1m -jar /Users/wimselles/Library/Android/sdk/build-tools/31.0.0-rc4/lib/apksigner.jar verify --print-certs /Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-debug-androidTest.apk
2021-06-02 09:16:24:096 [ADB] apksigner stdout: Signer #1 certificate DN: EMAILADDRESS=android@android.com, CN=Android, OU=Android, O=Android, L=Mountain View, ST=California, C=US
2021-06-02 09:16:24:096 [ADB] Signer #1 certificate SHA-256 digest: a40da80a59d170caa950cf15c18c454d47a39b26989d8b640ecd745ba71bf5dc
2021-06-02 09:16:24:096 [ADB] Signer #1 certificate SHA-1 digest: 61ed377e85d386a8dfee6b864bd85b0bfaa5af81
2021-06-02 09:16:24:096 [ADB] Signer #1 certificate MD5 digest: e89b158e4bcf988ebd09eb83f5378e87
2021-06-02 09:16:24:096 [ADB] 
2021-06-02 09:16:24:096 [ADB] sha256 hash did match for 'appium-uiautomator2-server-debug-androidTest.apk'
2021-06-02 09:16:24:097 [ADB] '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-debug-androidTest.apk' is signed with the default certificate
2021-06-02 09:16:24:097 [UiAutomator2] Server packages are not going to be (re)installed
2021-06-02 09:16:24:099 [UiAutomator2] Waiting up to 30000ms for services to be available
2021-06-02 09:16:24:099 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pm list instrumentation'
2021-06-02 09:16:24:120 [UiAutomator2] Instrumentation target 'io.appium.uiautomator2.server.test/androidx.test.runner.AndroidJUnitRunner' is available
2021-06-02 09:16:24:122 [ADB] Adding packages ["io.appium.settings","io.appium.uiautomator2.server","io.appium.uiautomator2.server.test"] to Doze whitelist
2021-06-02 09:16:24:122 [ADB] Got the following command chunks to execute: [["dumpsys","deviceidle","whitelist","+io.appium.settings",";","dumpsys","deviceidle","whitelist","+io.appium.uiautomator2.server",";","dumpsys","deviceidle","whitelist","+io.appium.uiautomator2.server.test",";"]]
2021-06-02 09:16:24:122 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys deviceidle whitelist +io.appium.settings ; dumpsys deviceidle whitelist +io.appium.uiautomator2.server ; dumpsys deviceidle whitelist +io.appium.uiautomator2.server.test ;'
2021-06-02 09:16:24:155 [UiAutomator2] No app capability. Assuming it is already on the device
2021-06-02 09:16:24:156 [ADB] Getting install status for com.android.chrome
2021-06-02 09:16:24:156 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package com.android.chrome'
2021-06-02 09:16:24:176 [ADB] 'com.android.chrome' is installed
2021-06-02 09:16:24:177 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell am force-stop com.android.chrome'
2021-06-02 09:16:24:203 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pm clear com.android.chrome'
2021-06-02 09:16:24:240 [AndroidDriver] Performed fast reset on the installed 'com.android.chrome' application (stop and clear)
2021-06-02 09:16:24:240 [UiAutomator2] Performing shallow cleanup of automation leftovers
2021-06-02 09:16:24:248 [UiAutomator2] No obsolete sessions have been detected (socket hang up)
2021-06-02 09:16:24:248 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell am force-stop io.appium.uiautomator2.server.test'
2021-06-02 09:16:24:290 [UiAutomator2] Starting UIAutomator2 server 4.21.1
2021-06-02 09:16:24:290 [UiAutomator2] Using UIAutomator2 server from '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-v4.21.1.apk' and test from '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-debug-androidTest.apk'
2021-06-02 09:16:24:290 [UiAutomator2] Waiting up to 30000ms for UiAutomator2 to be online...
2021-06-02 09:16:24:291 [ADB] Creating ADB subprocess with args: ["-P",5037,"-s","emulator-5554","shell","am","instrument","-w","-e","disableAnalytics",true,"io.appium.uiautomator2.server.test/androidx.test.runner.AndroidJUnitRunner"]
2021-06-02 09:16:24:297 [WD Proxy] Matched '/status' to command name 'getStatus'
2021-06-02 09:16:24:298 [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8200/wd/hub/status] with no body
2021-06-02 09:16:24:304 [WD Proxy] socket hang up
2021-06-02 09:16:24:738 [Instrumentation] io.appium.uiautomator2.server.test.AppiumUiAutomator2Server:
2021-06-02 09:16:25:308 [WD Proxy] Matched '/status' to command name 'getStatus'
2021-06-02 09:16:25:308 [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8200/wd/hub/status] with no body
2021-06-02 09:16:25:313 [WD Proxy] socket hang up
2021-06-02 09:16:26:314 [WD Proxy] Matched '/status' to command name 'getStatus'
2021-06-02 09:16:26:314 [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8200/wd/hub/status] with no body
2021-06-02 09:16:26:345 [WD Proxy] Got response with status 200: {"sessionId":"None","value":{"message":"UiAutomator2 Server is ready to accept commands","ready":true}}
2021-06-02 09:16:26:345 [UiAutomator2] The initialization of the instrumentation process took 2055ms
2021-06-02 09:16:26:345 [WD Proxy] Matched '/session' to command name 'createSession'
2021-06-02 09:16:26:345 [WD Proxy] Proxying [POST /session] to [POST http://127.0.0.1:8200/wd/hub/session] with body: {"capabilities":{"firstMatch":[{"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"browserName":"Chrome","platformName":"Android","deviceName":"Pixel_3_10.0","orientation":"PORTRAIT","platformVersion":"10.0"},"browserName":"Chrome","platformName":"Android","deviceName":"emulator-5554","orientation":"PORTRAIT","platformVersion":"10.0","appPackage":"com.android.chrome","appActivity":"com.google.android.apps.chrome.Main","deviceUDID":"emulator-5554"}],"alwaysMatch":{}}}
2021-06-02 09:16:26:357 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":{"capabilities":{"firstMatch":[{"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"browserName":"Chrome","platformName":"Android","deviceName":"Pixel_3_10.0","orientation":"PORTRAIT","platformVersion":"10.0"},"browserName":"Chrome","platformName":"Android","deviceName":"emulator-5554","orientation":"PORTRAIT","platformVersion":"10.0","appPackage":"com.android.chrome","appActivity":"com.google.android.apps.chrome.Main","deviceUDID":"emulator-5554"}],"alwaysMatch":{}},"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94"}}
2021-06-02 09:16:26:358 [WD Proxy] Determined the downstream protocol as 'W3C'
2021-06-02 09:16:26:366 [WD Proxy] Proxying [GET /appium/device/info] to [GET http://127.0.0.1:8200/wd/hub/session/3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94/appium/device/info] with no body
2021-06-02 09:16:26:382 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":{"androidId":"b78c9c3a30187984","apiVersion":"29","bluetooth":null,"brand":"google","carrierName":"Android","displayDensity":440,"locale":"en_US","manufacturer":"Google","model":"Android SDK built for x86","networks":[{"capabilities":{"SSID":null,"linkDownBandwidthKbps":1048576,"linkUpstreamBandwidthKbps":1048576,"networkCapabilities":"NET_CAPABILITY_NOT_METERED,NET_CAPABILITY_INTERNET,NET_CAPABILITY_NOT_RESTRICTED,NET_CAPABILITY_TRUSTED,NET_CAPABILITY_NOT_VPN,NET_CAPABILITY_VALIDATED,NET_CAPABILITY_NOT_ROAMING,NET_CAPABILITY_FOREGROUND,NET_CAPABILITY_NOT_CONGESTED,NET_CAPABILITY_NOT_SUSPENDED","signalStrength":-30,"transportTypes":"TRANSPORT_WIFI"},"detailedState":"CONNECTED","extraInfo":null,"isAvailable":true,"isConnected":true,"isFailover":false,"isRoaming":false,"state":"CONNECTED","subtype":0,"subtypeName":"","type":1,"typeName":"WIFI"}],"platformVersion":"10","realDisplaySize":"1080x2160","timeZone":"Europe/Amsterdam"}}
2021-06-02 09:16:26:382 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys window'
2021-06-02 09:16:26:409 [AndroidDriver] Screen already unlocked, doing nothing
2021-06-02 09:16:26:409 [AndroidDriver] Starting a chrome-based browser session
2021-06-02 09:16:26:419 [AndroidDriver] A port was not given, using random free port: 8000
2021-06-02 09:16:26:420 [AndroidDriver] Precalculated Chromedriver capabilities: {
2021-06-02 09:16:26:420 [AndroidDriver]   "androidPackage": "com.android.chrome",
2021-06-02 09:16:26:420 [AndroidDriver]   "androidDeviceSerial": "emulator-5554"
2021-06-02 09:16:26:420 [AndroidDriver] }
2021-06-02 09:16:26:420 [AndroidDriver] Before starting chromedriver, androidPackage is 'com.android.chrome'
2021-06-02 09:16:26:420 [Chromedriver] Changed state to 'starting'
2021-06-02 09:16:26:421 [Chromedriver] Using the static Chromedriver->Chrome mapping
2021-06-02 09:16:26:421 [Chromedriver] The most recent known Chrome version: 89.0.4389
2021-06-02 09:16:26:424 [Chromedriver] Found 1 executable in '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac'
2021-06-02 09:16:26:453 [Chromedriver] The following Chromedriver executables were found:
2021-06-02 09:16:26:453 [Chromedriver]     '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver_mac64_v74.0.3729.6' (version '74.0.3729.6', minimum Chrome version '74.0.3729')
2021-06-02 09:16:26:454 [ADB] Getting package info for 'com.android.chrome'
2021-06-02 09:16:26:454 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys package com.android.chrome'
2021-06-02 09:16:26:481 [Chromedriver] Found Chrome bundle 'com.android.chrome' version '74.0.3729'
2021-06-02 09:16:26:482 [Chromedriver] Found 1 executable capable of automating Chrome '74.0.3729'.
2021-06-02 09:16:26:482 [Chromedriver] Choosing the most recent, '/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver_mac64_v74.0.3729.6'.
2021-06-02 09:16:26:482 [Chromedriver] If a specific version is required, specify it with the `chromedriverExecutable`desired capability.
2021-06-02 09:16:26:483 [Chromedriver] Set chromedriver binary as: /Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver_mac64_v74.0.3729.6
2021-06-02 09:16:26:483 [Chromedriver] Killing any old chromedrivers, running: pkill -15 -f "/Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver_mac64_v74.0.3729.6.*--port=8000"
2021-06-02 09:16:26:521 [Chromedriver] No old chromedrivers seem to exist
2021-06-02 09:16:26:522 [Chromedriver] Cleaning this device's adb forwarded port socket connections: emulator-5554
2021-06-02 09:16:26:522 [ADB] List forwarding ports
2021-06-02 09:16:26:522 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 forward --list'
2021-06-02 09:16:26:530 [Chromedriver] Spawning chromedriver with: /Users/wimselles/.nvm/versions/node/v14.16.1/lib/node_modules/appium/node_modules/appium-chromedriver/chromedriver/mac/chromedriver_mac64_v74.0.3729.6 --url-base=wd/hub --port=8000 --adb-port=5037 --verbose
2021-06-02 09:16:26:546 [Chromedriver] Chromedriver version: '74.0.3729.6'
2021-06-02 09:16:26:547 [Chromedriver] Chromedriver v. 74.0.3729.6 does not fully support W3C protocol. Defaulting to MJSONWP
2021-06-02 09:16:26:547 [WD Proxy] Matched '/status' to command name 'getStatus'
2021-06-02 09:16:26:548 [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8000/wd/hub/status] with no body
2021-06-02 09:16:26:550 [WD Proxy] connect ECONNREFUSED 127.0.0.1:8000
2021-06-02 09:16:26:754 [WD Proxy] Matched '/status' to command name 'getStatus'
2021-06-02 09:16:26:754 [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8000/wd/hub/status] with no body
2021-06-02 09:16:26:764 [WD Proxy] Got response with status 200: {"sessionId":"","status":0,"value":{"build":{"version":"74.0.3729.6 (255758eccf3d244491b8a1317aa76e1ce10d57e9-refs/branch-heads/3729@{#29})"},"message":"ChromeDriver ready for new sessions.","os":{"arch":"x86_64","name":"Mac OS X","version":"10.16.0"},"ready":true}}
2021-06-02 09:16:26:765 [Chromedriver] Starting MJSONWP Chromedriver session with capabilities: {
2021-06-02 09:16:26:765 [Chromedriver]   "desiredCapabilities": {
2021-06-02 09:16:26:765 [Chromedriver]     "chromeOptions": {
2021-06-02 09:16:26:765 [Chromedriver]       "androidPackage": "com.android.chrome",
2021-06-02 09:16:26:765 [Chromedriver]       "androidDeviceSerial": "emulator-5554"
2021-06-02 09:16:26:765 [Chromedriver]     },
2021-06-02 09:16:26:765 [Chromedriver]     "loggingPrefs": {
2021-06-02 09:16:26:765 [Chromedriver]       "browser": "ALL"
2021-06-02 09:16:26:765 [Chromedriver]     }
2021-06-02 09:16:26:765 [Chromedriver]   }
2021-06-02 09:16:26:766 [Chromedriver] }
2021-06-02 09:16:26:766 [WD Proxy] Matched '/session' to command name 'createSession'
2021-06-02 09:16:26:766 [WD Proxy] Proxying [POST /session] to [POST http://127.0.0.1:8000/wd/hub/session] with body: {"desiredCapabilities":{"chromeOptions":{"androidPackage":"com.android.chrome","androidDeviceSerial":"emulator-5554"},"loggingPrefs":{"browser":"ALL"}}}
2021-06-02 09:16:28:024 [Chromedriver] Webview version: 'Chrome/74.0.3729.185'
2021-06-02 09:16:28:054 [WD Proxy] Got response with status 200: {"sessionId":"ba7844a1524fe6993d1fee7f425ff773","status":0,"value":{"acceptInsecureCerts":false,"acceptSslCerts":false,"applicationCacheEnabled":false,"browserConnectionEnabled":false,"browserName":"chrome","chrome":{"chromedriverVersion":"74.0.3729.6 (255758eccf3d244491b8a1317aa76e1ce10d57e9-refs/branch-heads/3729@{#29})"},"cssSelectorsEnabled":true,"databaseEnabled":false,"goog:chromeOptions":{"debuggerAddress":"localhost:49552"},"handlesAlerts":true,"hasTouchScreen":true,"javascriptEnabled":true,"locationContextEnabled":true,"mobileEmulationEnabled":false,"nativeEvents":true,"pageLoadStrategy":"normal","platform":"ANDROID","proxy":{},"rotatable":false,"setWindowRect":false,"strictFileInteractability":false,"takesHeapSnapshot":true,"takesScreenshot":true,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unexpectedAlertBehaviour":"ignore","version":"74.0.3729.185","webStorageEnabled":true}}
2021-06-02 09:16:28:054 [WD Proxy] Determined the downstream protocol as 'MJSONWP'
2021-06-02 09:16:28:054 [Chromedriver] Changed state to 'online'
2021-06-02 09:16:28:055 [UiAutomator2] Setting initial orientation to 'PORTRAIT'
2021-06-02 09:16:28:058 [WD Proxy] Matched '/orientation' to command name 'setOrientation'
2021-06-02 09:16:28:058 [WD Proxy] Proxying [POST /orientation] to [POST http://127.0.0.1:8200/wd/hub/session/3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94/orientation] with body: {"orientation":"PORTRAIT"}
2021-06-02 09:16:28:722 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":"PORTRAIT"}
2021-06-02 09:16:28:731 [WD Proxy] Proxying [GET /appium/device/pixel_ratio] to [GET http://127.0.0.1:8200/wd/hub/session/3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94/appium/device/pixel_ratio] with no body
2021-06-02 09:16:28:736 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":2.75}
2021-06-02 09:16:28:737 [WD Proxy] Matched '/appium/device/system_bars' to command name 'getSystemBars'
2021-06-02 09:16:28:737 [WD Proxy] Proxying [GET /appium/device/system_bars] to [GET http://127.0.0.1:8200/wd/hub/session/3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94/appium/device/system_bars] with no body
2021-06-02 09:16:28:742 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":{"statusBar":66}}
2021-06-02 09:16:28:742 [WD Proxy] Matched '/window/current/size' to command name 'getWindowSize'
2021-06-02 09:16:28:742 [WD Proxy] Proxying [GET /window/current/size] to [GET http://127.0.0.1:8200/wd/hub/session/3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94/window/current/size] with no body
2021-06-02 09:16:28:750 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":{"height":2028,"width":1080}}
2021-06-02 09:16:28:750 [Appium] New AndroidUiautomator2Driver session created successfully, session 48c98e90-2d93-469b-8e30-cb555a1fc56a added to master session list
2021-06-02 09:16:28:751 [BaseDriver] Event 'newSessionStarted' logged at 1622625388751 (11:16:28 GMT+0200 (Central European Summer Time))
2021-06-02 09:16:28:751 [W3C (48c98e90)] Cached the protocol value 'W3C' for the new session 48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:28:751 [W3C (48c98e90)] Responding to client with driver.createSession() result: {"capabilities":{"platform":"LINUX","webStorageEnabled":false,"takesScreenshot":true,"javascriptEnabled":true,"databaseEnabled":false,"networkConnectionEnabled":true,"locationContextEnabled":false,"warnings":{},"desired":{"browserName":"Chrome","platformName":"Android","deviceName":"Pixel_3_10.0","orientation":"PORTRAIT","platformVersion":"10.0"},"browserName":"Chrome","platformName":"Android","deviceName":"emulator-5554","orientation":"PORTRAIT","platformVersion":"10","appPackage":"com.android.chrome","appActivity":"com.google.android.apps.chrome.Main","deviceUDID":"emulator-5554","deviceApiLevel":29,"deviceScreenSize":"1080x2160","deviceScreenDensity":440,"deviceModel":"Android SDK built for x86","deviceManufacturer":"Google","pixelRatio":2.75,"statBarHeight":66,"viewportRect":{"left":0,"top":66,"width":1080,"height":1962}}}
2021-06-02 09:16:28:754 [HTTP] <-- POST /session 200 5989 ms - 899
2021-06-02 09:16:28:754 [HTTP] 
2021-06-02 09:16:28:825 [HTTP] --> POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/url
2021-06-02 09:16:28:825 [HTTP] {"url":"https://the-internet.herokuapp.com/login"}
2021-06-02 09:16:28:827 [W3C (48c98e90)] Driver proxy active, passing request on via HTTP proxy
2021-06-02 09:16:28:827 [WD Proxy] Matched '/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/url' to command name 'setUrl'
2021-06-02 09:16:28:827 [WD Proxy] Proxying [POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/url] to [POST http://127.0.0.1:8000/wd/hub/session/ba7844a1524fe6993d1fee7f425ff773/url] with body: {"url":"https://the-internet.herokuapp.com/login"}
2021-06-02 09:16:32:177 [WD Proxy] Got response with status 200: {"sessionId":"ba7844a1524fe6993d1fee7f425ff773","status":0,"value":null}
2021-06-02 09:16:32:177 [WD Proxy] Replacing sessionId ba7844a1524fe6993d1fee7f425ff773 with 48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:32:178 [HTTP] <-- POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/url 200 3353 ms - 65
2021-06-02 09:16:32:178 [HTTP] 
2021-06-02 09:16:32:182 [HTTP] --> POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element
2021-06-02 09:16:32:182 [HTTP] {"using":"css selector","value":"#username"}
2021-06-02 09:16:32:184 [W3C (48c98e90)] Driver proxy active, passing request on via HTTP proxy
2021-06-02 09:16:32:184 [WD Proxy] Matched '/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element' to command name 'findElement'
2021-06-02 09:16:32:184 [WD Proxy] Proxying [POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element] to [POST http://127.0.0.1:8000/wd/hub/session/ba7844a1524fe6993d1fee7f425ff773/element] with body: {"using":"css selector","value":"#username"}
2021-06-02 09:16:32:201 [WD Proxy] Got response with status 200: {"sessionId":"ba7844a1524fe6993d1fee7f425ff773","status":0,"value":{"ELEMENT":"0.305102702666354-1"}}
2021-06-02 09:16:32:201 [WD Proxy] Replacing sessionId ba7844a1524fe6993d1fee7f425ff773 with 48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:32:202 [HTTP] <-- POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element 200 19 ms - 154
2021-06-02 09:16:32:202 [HTTP] 
2021-06-02 09:16:32:212 [HTTP] --> POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element/0.305102702666354-1/click
2021-06-02 09:16:32:212 [HTTP] {}
2021-06-02 09:16:32:213 [W3C (48c98e90)] Driver proxy active, passing request on via HTTP proxy
2021-06-02 09:16:32:214 [WD Proxy] Matched '/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element/0.305102702666354-1/click' to command name 'click'
2021-06-02 09:16:32:214 [WD Proxy] Proxying [POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element/0.305102702666354-1/click] to [POST http://127.0.0.1:8000/wd/hub/session/ba7844a1524fe6993d1fee7f425ff773/element/0.305102702666354-1/click] with body: {}
2021-06-02 09:16:32:281 [WD Proxy] Got response with status 200: {"sessionId":"ba7844a1524fe6993d1fee7f425ff773","status":0,"value":null}
2021-06-02 09:16:32:281 [WD Proxy] Replacing sessionId ba7844a1524fe6993d1fee7f425ff773 with 48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:32:281 [HTTP] <-- POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/element/0.305102702666354-1/click 200 69 ms - 65
2021-06-02 09:16:32:281 [HTTP] 
2021-06-02 09:16:32:286 [HTTP] --> POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:286 [HTTP] {"actions":[{"type":"key","id":"keyboard","actions":[{"type":"keyDown","value":"t"},{"type":"keyDown","value":"o"},{"type":"keyDown","value":"m"},{"type":"keyDown","value":"s"},{"type":"keyDown","value":"m"},{"type":"keyDown","value":"i"},{"type":"keyDown","value":"t"},{"type":"keyDown","value":"h"},{"type":"keyUp","value":"t"},{"type":"keyUp","value":"o"},{"type":"keyUp","value":"m"},{"type":"keyUp","value":"s"},{"type":"keyUp","value":"m"},{"type":"keyUp","value":"i"},{"type":"keyUp","value":"t"},{"type":"keyUp","value":"h"}]}]}
2021-06-02 09:16:32:287 [W3C (48c98e90)] Driver proxy active, passing request on via HTTP proxy
2021-06-02 09:16:32:288 [WD Proxy] Matched '/session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions' to command name 'performActions'
2021-06-02 09:16:32:289 [WD Proxy] Proxying [POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions] to [POST http://127.0.0.1:8000/wd/hub/session/ba7844a1524fe6993d1fee7f425ff773/actions] with body: {"actions":[{"type":"key","id":"keyboard","actions":[{"type":"keyDown","value":"t"},{"type":"keyDown","value":"o"},{"type":"keyDown","value":"m"},{"type":"keyDown","value":"s"},{"type":"keyDown","value":"m"},{"type":"keyDown","value":"i"},{"type":"keyDown","value":"t"},{"type":"keyDown","value":"h"},{"type":"keyUp","value":"t"},{"type":"keyUp","value":"o"},{"type":"keyUp","value":"m"},{"type":"keyUp","value":"s"},{"type":"keyUp","value":"m"},{"type":"keyUp","value":"i"},{"type":"keyUp","value":"t"},{"type":"keyUp","value":"h"}]}]}
2021-06-02 09:16:32:654 [WD Proxy] Got response with status 200: {"sessionId":"ba7844a1524fe6993d1fee7f425ff773","status":0,"value":null}
2021-06-02 09:16:32:654 [WD Proxy] Replacing sessionId ba7844a1524fe6993d1fee7f425ff773 with 48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:32:654 [HTTP] <-- POST /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions 200 368 ms - 65
2021-06-02 09:16:32:655 [HTTP] 
2021-06-02 09:16:32:659 [HTTP] --> DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:659 [HTTP] {}
2021-06-02 09:16:32:666 [HTTP] No route found for /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:667 [HTTP] <-- DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions 404 8 ms - 262
2021-06-02 09:16:32:667 [HTTP] 
2021-06-02 09:16:32:677 [HTTP] --> DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:677 [HTTP] {}
2021-06-02 09:16:32:686 [HTTP] No route found for /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:687 [HTTP] <-- DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions 404 10 ms - 262
2021-06-02 09:16:32:687 [HTTP] 
2021-06-02 09:16:32:689 [HTTP] --> DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:690 [HTTP] {}
2021-06-02 09:16:32:690 [HTTP] No route found for /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:691 [HTTP] <-- DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions 404 1 ms - 262
2021-06-02 09:16:32:691 [HTTP] 
2021-06-02 09:16:32:693 [HTTP] --> DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:693 [HTTP] {}
2021-06-02 09:16:32:693 [HTTP] No route found for /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions
2021-06-02 09:16:32:694 [HTTP] <-- DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a/actions 404 1 ms - 262
2021-06-02 09:16:32:694 [HTTP] 
2021-06-02 09:16:32:702 [HTTP] --> DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a
2021-06-02 09:16:32:702 [HTTP] {}
2021-06-02 09:16:32:702 [W3C (48c98e90)] Calling AppiumDriver.deleteSession() with args: ["48c98e90-2d93-469b-8e30-cb555a1fc56a"]
2021-06-02 09:16:32:702 [BaseDriver] Event 'quitSessionRequested' logged at 1622625392702 (11:16:32 GMT+0200 (Central European Summer Time))
2021-06-02 09:16:32:702 [Appium] Removing session 48c98e90-2d93-469b-8e30-cb555a1fc56a from our master session list
2021-06-02 09:16:32:703 [UiAutomator2] Deleting UiAutomator2 session
2021-06-02 09:16:32:703 [AndroidDriver] Stopping chromedriver for context CHROMIUM
2021-06-02 09:16:32:704 [Chromedriver] Changed state to 'stopping'
2021-06-02 09:16:32:705 [WD Proxy] Proxying [DELETE /] to [DELETE http://127.0.0.1:8000/wd/hub/session/ba7844a1524fe6993d1fee7f425ff773] with no body
2021-06-02 09:16:32:855 [WD Proxy] Got response with status 200: {"sessionId":"ba7844a1524fe6993d1fee7f425ff773","status":0,"value":null}
2021-06-02 09:16:32:861 [Chromedriver] Changed state to 'stopped'
2021-06-02 09:16:32:861 [UiAutomator2] Deleting UiAutomator2 server session
2021-06-02 09:16:32:861 [WD Proxy] Matched '/' to command name 'deleteSession'
2021-06-02 09:16:32:861 [WD Proxy] Proxying [DELETE /] to [DELETE http://127.0.0.1:8200/wd/hub/session/3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94] with no body
2021-06-02 09:16:32:877 [WD Proxy] Got response with status 200: {"sessionId":"3b5f20b4-f0e7-4fbf-8984-9ba3fdb19e94","value":null}
2021-06-02 09:16:32:877 [Logcat] Stopping logcat capture
2021-06-02 09:16:32:880 [ADB] Removing forwarded port socket connection: 8200 
2021-06-02 09:16:32:880 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 forward --remove tcp:8200'
2021-06-02 09:16:32:894 [UiAutomator2] Restoring hidden api policy to the device default configuration
2021-06-02 09:16:32:895 [ADB] Running '/Users/wimselles/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy''
2021-06-02 09:16:33:131 [BaseDriver] Event 'quitSessionFinished' logged at 1622625393131 (11:16:33 GMT+0200 (Central European Summer Time))
2021-06-02 09:16:33:132 [W3C (48c98e90)] Received response: null
2021-06-02 09:16:33:132 [W3C (48c98e90)] But deleting session, so not returning
2021-06-02 09:16:33:132 [W3C (48c98e90)] Responding to client with driver.deleteSession() result: null
2021-06-02 09:16:33:134 [HTTP] <-- DELETE /session/48c98e90-2d93-469b-8e30-cb555a1fc56a 200 432 ms - 14
2021-06-02 09:16:33:134 [HTTP] 
2021-06-02 09:16:33:254 [Appium] Received SIGTERM - shutting down
```

**Comment by christian-bromann** [comment]
@Sam55555 I recommend not to run tests over JWP given it is a deprecated protocol, especially if you want to test on the latest devices.

**Comment by Sam55555** [comment]
@christian-bromann according to Saucelabs support, Saucelabs supports only JWT for mobile simulators. And with W3C it cause error from the previous two posts about Appium.

**Comment by christian-bromann** [comment]
Created a new issue #6981
